community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/idp/IDPQueryGraphSolver.scala

```diff
@@ -43,6 +40,10 @@ trait IDPQueryGraphSolverMonitor extends IDPSolverMonitor {
   def endConnectingComponents(graph: QueryGraph, result: LogicalPlan): Unit
 }
 
+object IDPQueryGraphSolver {
+  val VERBOSE = java.lang.Boolean.getBoolean("pickBestPlan.VERBOSE")
```

Comment:
This object and value is used in only one place, and not in a tight inner loop, so I see no reason to have an object dedicated to this purpose. Perhaps simpler (ie. less code) to only get the system property at the point it is used?